lib/perl/Genome/Sys/LockMigrationProxy.pm

```diff
@@ -0,0 +1,100 @@
+package Genome::Sys::LockMigrationProxy;
+
+use strict;
+use warnings;
+
+use Carp qw(croak);
+use Genome::Sys::LockProxy qw();
+use Params::Validate qw(validate validate_with HASHREF);
+
+=item new()
+
+Keyword Arguments:
+
+=over
+
+=item old
+
+A hash reference containing the constructor parameters for the existing/old lock.
+
+=item new
+
+A hash reference containing the constructor parameters for the desired/new lock.
+
+=cut
+
+sub new {
+    my $class = shift;
+    my %params = validate(@_, {
+        old => { type => HASHREF },
+        new => { type => HASHREF },
+    });
+
+    my $primoridal_ooze = {
+        old => Genome::Sys::LockProxy->new(%{$params{old}}),
+        new => Genome::Sys::LockProxy->new(%{$params{new}}),
+    };
+
+    return bless $primoridal_ooze, $class;
+}
+
+=item lock()
+
+C<lock()> is an instance method with optional parameters C<block_sleep>,
+C<max_try>, or C<wait_announce_interval>.  It behaves similar to
+C<Genome::Sys::Lock::lock_resource()> but on sucess C<lock()> returns an
+instance of C<Genome::Sys::LockMigrationProxy> that can later be unlocked with
+C<unlock()>.  Both the C<old> lock and the C<new> lock will be acquired, in
+that order, or neither will.
+
+=cut
+
+sub lock {
+    my $self = shift;
+    my %params = validate_with(
+        params => \@_,
+        spec => {
+            block_sleep => 0,
+            max_try => 0,
+            wait_announce_interval => 0,
+        },
```

Comment:
Could this one also benefit from `Genome::Sys::Lock->LOCK_RESOURCE_PARAMS_SPEC`?